src/inc/genXplatEtw.pl

```diff
@@ -0,0 +1,782 @@
+
+            #generate EventEnabled
+            print lTTngImpl "extern \"C\" BOOL  EventEnabled$eventName(){ return TRUE;}\n";
+            #generate FireEtw functions
+            my $fnptype = "extern \"C\" ULONG  FireEtw$eventName(\n";
+            my $linefnptype="";
+
+            if ( $templateName ne "")
+            {
+                foreach my $subtemplates (@{$allTemplatemap{$templateName}})
+                {
+                    my $fnSignature = $allLTTngTemplateTypes{$subtemplates} ;
+
+
+                    foreach  my $params (@$fnSignature)
+                    {
+                        my $wintypeName = $params->{"type"};
+                        my $typewName = $palDataTypeMapping{$wintypeName} || die "no mapping for $wintypeName";
+                        my $winCount = $params->{"count"};
+                        my $countw = $palDataTypeMapping{$winCount} || die "no mapping for $winCount";
+                        $linefnptype=$linefnptype.$lindent.$typewName.$countw." ";
+                        $linefnptype=$linefnptype.$params->{"var"};
+                        $linefnptype=$linefnptype.",\n";
+                    }
+
+                }
+                $linefnptype =~ s/,+$//;
+            }
+            $fnptype = $fnptype.$linefnptype.")\n";
+            $fnptype = $fnptype."{\nULONG Error = ERROR_WRITE_FAULT;\n";
+            print lTTngImpl $fnptype;
+#start of fn body
+            my $linefnbody="";
+            if ($templateName ne '')
+            {
+                foreach my $subtemplates (@{$allTemplatemap{$templateName}})
+                {
+                    my $fnSignature = $allLTTngTemplateTypes{$subtemplates};
+                    my  $subevent="";
+
+                    if ($linefnbody ne "")
+                    {
+                        $linefnbody =~ s/,+$//;
+                        $linefnbody =$linefnbody."        );\n";
+                    }
+
+                    $subevent=$subtemplates;
+                    $subevent=~s/$templateName//;
+                    $linefnbody=<<FN_body_cont;
+                    $linefnbody
+     tracepoint(
+        $providerName,
+        $eventName$subevent,
+FN_body_cont
+                    foreach  my $params (@$fnSignature)
+                    {
+                        $linefnbody =$linefnbody."        ";
+                        my $wintypeName = $params->{"type"};
+                        my $winCount = $params->{"count"};
+                        my $paramname = $params->{"var"};
+                        my $ctf_type =$ctfDataTypeMapping{$winCount} || $ctfDataTypeMapping{$wintypeName} || die "no such ctf mapping for  $eventName: $wintypeName $paramname "; ;
+                        if ($ctf_type eq "ctf_string")
+                        {
+                            if($wintypeName eq "win:UnicodeString")
+                            {
+                                #emit code to convert unicode to ansi string
```

Comment:
WideCharToMultiByte is a function which converts UTF-16 strings to ASCII, which is in turn passed to LTTng